js/APIMiddleware.js

```diff
@@ -0,0 +1,42 @@
+let result;
+/**
+ * API - Description
+ *
+ * @returns {type} Description
+ */
+const API = {
+    get: async (url) => {
+        await fetch(url)
+            .then(response => response.json())
+            .then((json) => {
+                if (!json.error) {
+                    result = json;
+                } else {
+                    if (json.info == "token failed to validate") {
+                        localStorage.removeItem('token');
+                        document.location.href = "index.html";
+                    } else {
+                        console.log(json);
```

Comment:
If there is an error that's not token related `result` will be undefined but still retruned. No useful data will be returned. There are several errors that are needed to notify the user about that is now supressed and only vissible in the console. It is recommended to send back some kind of indicator of an error. The code invoking the call can then take care of it and notify the user relevant information depending on context.

Reply:
should I just send back json when else is true instead? json contains useful data from backend that the code invoking the call can handle

Reply:
needs a second look

Reply:
Looks good, the code calling this function will have to handle the response and decide what it wants to do with the errors.